webapp/src/blocks.tsx

```diff
@@ -532,31 +548,34 @@ export class Editor extends toolboxeditor.ToolboxEditor {
         };
     }
 
-    // private initAccessibleBlocks() {
-    //     const enabled = pxt.appTarget.appTheme?.accessibleBlocks;
-    //     if (enabled && !this.navigationController) {
-    //         this.navigationController = new NavigationController() as any;
-
-    //         this.navigationController.init();
-    //         this.navigationController.addWorkspace(this.editor);
-
-    //         (Navigation as any).prototype.focusToolbox = (workspace: Blockly.WorkspaceSvg) => {
-    //             const toolbox = this.toolbox;
-    //             if (!toolbox) return;
-    //             this.focusToolbox();
-    //             this.navigationController.navigation.resetFlyout(workspace, false);
-    //             this.navigationController.navigation.setState(workspace, "toolbox");
-    //         }
-    //     }
-    // }
-
-    // public enableAccessibleBlocks(enable: boolean) {
-    //     if (enable) {
-    //         this.navigationController.enable(this.editor);
-    //     } else {
-    //         this.navigationController.disable(this.editor);
-    //     }
-    // }
+    private initAccessibleBlocks() {
+        const enabled = pxt.appTarget.appTheme?.accessibleBlocks;
+        if (enabled && !this.keyboardNavigation) {
+            this.keyboardNavigation = new KeyboardNavigation(this.editor);
+
+            const injectionDiv = document.getElementById('blocksEditor');
+            injectionDiv.classList.add("accessibleBlocks");
+            const focusRingDiv = injectionDiv.appendChild(document.createElement("div"))
+            focusRingDiv.className = "blocklyWorkspaceFocusRingLayer";
+            this.editor.getSvgGroup().addEventListener("focus", () => {
+                focusRingDiv.dataset.focused = "true";
+            })
+            this.editor.getSvgGroup().addEventListener("blur", () => {
+                delete focusRingDiv.dataset.focused;
+            })
```

Comment:
This gives a more reasonable focus ring for the workspace, one that isn't occluded on the left hand side by the toolbox. We should discuss this with Blockly and potentially move the behaviour to the plugin (maybe with options for a better implementation).

Reply:
We are continuing to have discussions regarding the focus indicators, but we propose moving forward with this change for this initial PR.